com.unity.render-pipelines.universal/Runtime/2D/LightCachedMeshData.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Collections;
+using Unity.Mathematics;
+using UnityEngine.Experimental.Rendering.Universal.LibTessDotNet;
+using UnityEngine.Rendering;
+using UnityEngine.U2D;
+
+namespace UnityEngine.Experimental.Rendering.Universal
+{
+
+    // Simple Cache for Shape Geometry Data.
+    // Note: The cache only gets saved in Editor mode.
+    // On Runtime, data is only used on Load. Any further changes to Light2D will involve regeneration.
+    [AddComponentMenu("")]
+    internal class LightCachedMeshData : MonoBehaviour
+    {
+
+        [SerializeField] [HideInInspector] LightUtility.ParametricLightMeshVertex[] m_VertexArray = null;
+        [SerializeField] [HideInInspector] ushort[] m_IndexArray = null;
+        bool m_RequiresUpload = true;
+
+        internal ushort[] indices => m_IndexArray;
+        internal LightUtility.ParametricLightMeshVertex[] vertices => m_VertexArray;
+        internal bool allowUpdateOnTests = false; // Only Used during Playmode Tests.
+        internal bool requiresUpload { get => m_RequiresUpload; set => m_RequiresUpload = value; }
+
+        // Has Geometry Data
+        internal bool HasGeometryData()
+        {
+            return null != m_IndexArray && null != m_VertexArray && m_IndexArray.Length != 0 && m_VertexArray.Length != 0;
+        }
+
+        // Has Uploaded.
+        internal bool RequiresUpload()
+        {
+            return m_RequiresUpload && HasGeometryData();
+        }
+
+        // Set Geometry Cache.
+        internal void SetGeometryCache(LightUtility.ParametricLightMeshVertex[] vertexArray, ushort[] indexArray)
+        {
+#if UNITY_EDITOR
```

Comment:
Why not ifdef the entire method?

Reply:
Fixed in 94f30ac326f1fc161502821c5d49373b19127b16